lib/message_bus/rack/middleware.rb

```diff
@@ -242,4 +248,18 @@ def start_listener
       end
     end
   end
+
+  # @param query_string [String]
+  # @return [String]
+  def query_params(query_string)
+    Rack::Utils.default_query_parser.parse_nested_query(query_string, '&')
+  end
+
+  # @param env [Hash]
+  # @return [Array<Array<String, Proc>>]
+  def compute_message_filters(env)
+    @bus.client_message_filters.map do |channel_name, filter|
+      [channel_name, filter.curry(2).call(query_params(env['QUERY_STRING']))]
```

Comment:
why you said here filter.curry(2) and than you call with one argument?

Reply:
`filter` proc should accept two arguments. If user defines it as follows:

```ruby
filter_proc = proc {|params, message|  do_something }
```

Then it is just fine and we can immediately bind `params` and return new proc as follows:
```ruby
bound = filter.curry[params]
```

Later we can use it as follows:

```ruby
bound.curry[message] # fill evaluate our filter proc with query params and a message
```

But if user defines it as follows:

```ruby
pr = proc {|*args| params = args[0]; message = args[1] }
```

we will have a problems binding `params` only. Ss stated in docs:

```
The optional arity argument should be supplied when currying procs with variable arguments to determine how many arguments are needed before the proc is called.
```

Thus, I need to explicitly say that a proc accepts two arguments, and then bind first argument(query params).

Refs is [here](https://ruby-doc.org/3.2.2/Proc.html#method-i-curry)